scripts/operations/ceph/enable-smart-mon-storage-nodes.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+smartmon_url=''
+ssh_options="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+
+function get_smartmon_url() {
+  local repos
+
+  if ! curl -sSfk https://packages.local/service/rest/v1/repositories >&/dev/null; then
+    echo "***"
+    echo "WARNING: Unable to contact Nexus! This must be addressed before proceeding."
+    echo "***"
+    return 1
+  fi
+
+  # get a list of repos that start with "csm-noos"
+  if IFS=$'\n' read -rd '' -a repos; then
+        :
+  fi <<< "$(curl -sSfk https://packages.local/service/rest/v1/repositories | jq -r '.[] | .["name"]' | grep ^csm-noos | tr '\n' ' ')"
+
+  echo "Will look for smart-mon rpm in the following repos: ${repos[@]}"
+
+  # search through the csm-noos repos looking for our package
+  for repo in "${repos[@]}"; do
+    # Retrieve the packages from nexus
+    if [[ -z "$smartmon_url" ]]; then
+      smartmon_url=$(paginate "https://packages.local/service/rest/v1/components?repository=$repo" \
+        | jq -r '.items[] | .assets[] | .downloadUrl' | grep smart-mon | sort -V | tail -1)
+      if [[ $? -ne 0 ]]; then
+        return 1
+      fi
```

Comment:
Same comment as https://github.com/Cray-HPE/docs-csm/pull/4299/files#r1337850032.
```suggestion
      if ! smartmon_url="$(paginate "https://packages.local/service/rest/v1/components?repository=$repo" \
        | jq -r '.items[] | .assets[] | .downloadUrl' | grep smart-mon | sort -V | tail -1 )"; then return 1; fi
        return 1
      fi
```